<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html>
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Function reference â€¢ BaMORC</title>

<!-- jquery -->
<script src="https://code.jquery.com/jquery-3.1.0.min.js" integrity="sha384-nrOSfDHtoPMzJHjVTdCopGqIqeYETSXhZDFyniQ8ZHcVy08QesyHcnOUpMpqnmWq" crossorigin="anonymous"></script>
<!-- Bootstrap -->
<link href="https://maxcdn.bootstrapcdn.com/bootswatch/3.3.7/cosmo/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">

<script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha384-Tc5IQib027qvyjSMfHjOMaLkfuWVxZxUPnCJA7l2mCWNIpG9mGCD8wGNIcPD7Txa" crossorigin="anonymous"></script>

<!-- Font Awesome icons -->
<link href="https://maxcdn.bootstrapcdn.com/font-awesome/4.6.3/css/font-awesome.min.css" rel="stylesheet" integrity="sha384-T8Gy5hrqNKT+hzMclPo118YTQO6cYprQmhrYwIiQ/3axmI1hQomh7Ud2hPOy8SP1" crossorigin="anonymous">

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/1.7.1/clipboard.min.js" integrity="sha384-cV+rhyOuRHc9Ub/91rihWcGmMmCXDeksTtCihMupQHSsi8GIIRDG0ThDc3HGQFJ3" crossorigin="anonymous"></script>

<!-- sticky kit -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/sticky-kit/1.1.3/sticky-kit.min.js" integrity="sha256-c4Rlo1ZozqTPE2RLuvbusY3+SU1pQaJC0TjuhygMipw=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>



<meta property="og:title" content="Function reference" />

<meta property="og:image" content="/logo.png" />
<meta name="twitter:card" content="summary" />



<!-- mathjax -->
<script src='https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML'></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->


  </head>

  <body>
    <div class="container template-reference-index">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">BaMORC</a>
        <span class="label label-default" data-toggle="tooltip" data-placement="bottom" title="Released package">1.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../articles/quickstart.html">Quickstart</a>
</li>
<li>
  <a href="../articles/CLI_guide.html">CLI Guide</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li>
  <a href="../news/index.html">ChangeLog</a>
</li>
      </ul>
      
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/billchenxi/BaMORC">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      
      </header>

<div class="row">
  <div class="contents col-md-9">
    <div class="page-header">
      <h1>Reference</h1>
    </div>

    <table class="ref-index">

    <colgroup>
      
      <col class="alias" />
      <col class="title" />
    </colgroup>

    <tbody>
      <tr>
        <th colspan="2">
          <h2 id="section-data-processing-functions" class="hasAnchor"><a href="#section-data-processing-functions" class="anchor"></a>Data Processing Functions</h2>
          <p class="section-desc"><p>BaMORC currently support file delimiters of comma, semicolon and whitespace. For users who want to run analysis on existing dataset from BMRB (NMR-STAR version 3), they can use either the read_NMRSTAR_file function with parameter of local file path or the readDBFile function with parameters of BMRB ID and flag that indicates whether data is assigned or unassigned. If readDBFile is used, BaMORC will utilize the BMRB web API to fetch the corresponding BMRB entry matching the ID.</p></p>
        </th>
      </tr>
      <tr>
        
        <td>
          <p><code><a href="read_db_file.html">read_db_file()</a></code> </p>
        </td>
        <td><p><code>read_db_file()</code> reads in data from existing database that included in the BaMORC package. This database was extracted from RefDB database.</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="read_nmrstar_file.html">read_nmrstar_file()</a></code> </p>
        </td>
        <td><p>Extracts data from BMRB STAR 3.0 file.
<code>read_nmrstar_file()</code> parses BMRB STAR 3.0 file. It will extract sequence information and chemical shifts for both alpha and beta carbons.</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="read_raw_file.html">read_raw_file()</a></code> </p>
        </td>
        <td><p>Extracts data from a protein NMR experimental peak list.
<code>read_raw_file()</code> function reads in a user provided protein NMR experimental peak list. It currently supports file format in csv, txt with deliminator of comma, whitespace or semicolon.
Note: please don't leave space between sequence and chemical shifts data, otherwise it will report error.</p></td>
      </tr>
    </tbody><tbody>
      <tr>
        <th colspan="2">
          <h2 id="section-calculation-functions" class="hasAnchor"><a href="#section-calculation-functions" class="anchor"></a>Calculation Functions</h2>
          <p class="section-desc"><p>Funcations calculate necessary statsitical values that essential for BaMORC analysis.</p></p>
        </th>
      </tr>
      <tr>
        
        <td>
          <p><code><a href="calculate_aa_prob.html">calculate_aa_prob()</a></code> </p>
        </td>
        <td><p>Calculates an amino acid typing probability.</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="calculate_chi_squared_stat.html">calculate_chi_squared_stat()</a></code> </p>
        </td>
        <td><p>Calculates a chi squared statistic(s).</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="calculate_mse.html">calculate_mse()</a></code> </p>
        </td>
        <td><p>Calculates mean squared error</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="calculate_rcf.html">calculate_rcf()</a></code> </p>
        </td>
        <td><p>Calculates the relative cumulative frequency for amino acid and secondary structure.</p></td>
      </tr>
    </tbody><tbody>
      <tr>
        <th colspan="2">
          <h2 id="section-bamorc-analysis-functions" class="hasAnchor"><a href="#section-bamorc-analysis-functions" class="anchor"></a>BaMORC Analysis Functions</h2>
          <p class="section-desc"><p>Fundamental core function for BaMORC analysis, which requires the secondary structure information.</p></p>
        </th>
      </tr>
      <tr>
        
        <td>
          <p><code><a href="bamorc.html">bamorc()</a></code> </p>
        </td>
        <td><p>Calculates the referencing correction value.</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="unassigned_bamorc.html">unassigned_bamorc()</a></code> </p>
        </td>
        <td><p>Calculates the referencing correction value for unassigned protein NMR peaklists.
<code>unassigned_bamorc()</code> will analyze unassigned protein NMR spectra, first groups the peaklist via SSC, then estimates the secondary structure via JPred, finally using BaMORC core function to calculate the reference correction value.</p></td>
      </tr>
    </tbody><tbody>
      <tr>
        <th colspan="2">
          <h2 id="section-utility-functions" class="hasAnchor"><a href="#section-utility-functions" class="anchor"></a>Utility Functions</h2>
          <p class="section-desc"><p>Utility functions used by BaMORC analysis.</p></p>
        </th>
      </tr>
      <tr>
        
        <td>
          <p><code><a href="jpred_fetcher.html">jpred_fetcher()</a></code> </p>
        </td>
        <td><p>Using JPred Mass-submission scheduler program to submit protein sequence and return secondary structure results.</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="sample_data_generator.html">sample_data_generator()</a></code> </p>
        </td>
        <td><p>sample_data_generator</p></td>
      </tr>
    </tbody><tbody>
      <tr>
        <th colspan="2">
          <h2 id="section-data-objects" class="hasAnchor"><a href="#section-data-objects" class="anchor"></a>Data Objects</h2>
          <p class="section-desc"></p>
        </th>
      </tr>
      <tr>
        
        <td>
          <p><code><a href="AA57OLMatrix.html">AA57OLMatrix</a></code> </p>
        </td>
        <td><p>Overlapping Matrix for 57 amino acid and secondary structure combinations.</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="AA57OLWeights.html">AA57OLWeights</a></code> </p>
        </td>
        <td><p>Overlapping weights for 57 amino acid and secondary structure combinations.</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="CAMuTable.html">CAMuTable</a></code> </p>
        </td>
        <td><p>Mean chemical shift values for alpha carbon .</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="CASdTable.html">CASdTable</a></code> </p>
        </td>
        <td><p>Standard deviation of chemical shift values for alpha carbon .</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="CBMuTable.html">CBMuTable</a></code> </p>
        </td>
        <td><p>Mean chemical shift values for beta carbon .</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="CBSdTable.html">CBSdTable</a></code> </p>
        </td>
        <td><p>Standard deviation of chemical shift values for beta carbon .</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="AvgCov.t.html">AvgCov.t</a></code> </p>
        </td>
        <td><p>Average covariance across three secondary structures for all amino acid typings (including oxidized cystine).</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="CarbonCov.t.html">CarbonCov.t</a></code> </p>
        </td>
        <td><p>Covariance values of chemical shifts of alpha and beta carbons.</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="RefDB.StatCA.html">RefDB.StatCA</a></code> </p>
        </td>
        <td><p>Statistics of chemical shifts values of alpha carbons from RefDB.</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="RefDB.StatCB.html">RefDB.StatCB</a></code> </p>
        </td>
        <td><p>Statistics of chemical shifts values of beta carbons from RefDB.</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="RefDB_data.html">RefDB_data</a></code> </p>
        </td>
        <td><p>RefDB object</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="inverseMatrices.html">inverseMatrices</a></code> </p>
        </td>
        <td><p>inverseMatrices.</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="aaCodes1Letter19.html">aaCodes1Letter19</a></code> </p>
        </td>
        <td><p>Single-letter amino acid naming convention.</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="aaCodes1Letter20.html">aaCodes1Letter20</a></code> </p>
        </td>
        <td><p>Single-letter amino acid naming convention.</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="aaCodes3Letter1stCap.html">aaCodes3Letter1stCap</a></code> </p>
        </td>
        <td><p>Three-letter amino acid naming convention with first letter capitalized.</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="aaCodes3LetterAllCap.v.html">aaCodes3LetterAllCap.v</a></code> </p>
        </td>
        <td><p>Three-letter amino acid naming convention with all letters capitalized.</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="ID.html">ID</a></code> </p>
        </td>
        <td><p>RefDB ID included in the BaMORC package.</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="cname.html">cname</a></code> </p>
        </td>
        <td><p>All the amino acids and secondary structures combinations for easy access.</p></td>
      </tr>
    </tbody><tbody>
      <tr>
        <th colspan="2">
          <h2 id="section-built-in-samples" class="hasAnchor"><a href="#section-built-in-samples" class="anchor"></a>Built-in Samples</h2>
          <p class="section-desc"></p>
        </th>
      </tr>
      <tr>
        
        <td>
          <p><code><a href="chemicalShifts.html">chemicalShifts</a></code> </p>
        </td>
        <td><p>Pre-defined sample chemical shifts data.</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="aaFreq.html">aaFreq</a></code> </p>
        </td>
        <td><p>Pre-defined amino acid frequency data.</p></td>
      </tr>
    </tbody>
    </table>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
    <h2>Contents</h2>
    <ul class="nav nav-pills nav-stacked">
      <li><a href="#section-data-processing-functions">Data Processing Functions</a></li>
      <li><a href="#section-calculation-functions">Calculation Functions</a></li>
      <li><a href="#section-bamorc-analysis-functions">BaMORC Analysis Functions</a></li>
      <li><a href="#section-utility-functions">Utility Functions</a></li>
      <li><a href="#section-data-objects">Data Objects</a></li>
      <li><a href="#section-built-in-samples">Built-in Samples</a></li>
    </ul>
  </div>
</div>

      <footer>
      <div class="copyright">
  <p>Developed by Xi Chen, Andrey Smelter, Hunter Moseley.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="http://pkgdown.r-lib.org/">pkgdown</a>.</p>
</div>

      </footer>
   </div>

  

  </body>
</html>

